packages/block-library/src/navigation-menu-item/edit.js

```diff
@@ -1,93 +1,81 @@
-/**
- * External dependencies
- */
-import { invoke } from 'lodash';
-
 /**
  * WordPress dependencies
  */
 import { withSelect } from '@wordpress/data';
 import {
-	Dropdown,
 	ExternalLink,
-	IconButton,
 	PanelBody,
 	TextareaControl,
 	TextControl,
+	Toolbar,
 	ToggleControl,
+	ToolbarButton,
 } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import {
+	BlockControls,
 	InnerBlocks,
 	InspectorControls,
 	PlainText,
+	URLPopover,
 } from '@wordpress/block-editor';
 import {
 	Fragment,
-	useCallback,
 	useRef,
+	useState,
 } from '@wordpress/element';
 
-/**
- * Internal dependencies
- */
-import MenuItemActions from './menu-item-actions';
-const POPOVER_PROPS = { noArrow: true };
-
 function NavigationMenuItemEdit( {
 	attributes,
-	clientId,
 	isSelected,
 	isParentOfSelectedBlock,
 	setAttributes,
 } ) {
 	const plainTextRef = useRef( null );
-	const onEditLableClicked = useCallback(
-		( onClose ) => () => {
-			onClose();
-			invoke( plainTextRef, [ 'current', 'textarea', 'focus' ] );
-		},
-		[ plainTextRef ]
-	);
+	const [ isLinkOpen, setLinkOpen ] = useState( false );
+	const { label, url } = attributes;
 	let content;
 	if ( isSelected ) {
 		content = (
 			<div className="wp-block-navigation-menu-item__edit-container">
 				<PlainText
 					ref={ plainTextRef }
 					className="wp-block-navigation-menu-item__field"
-					value={ attributes.label }
-					onChange={ ( label ) => setAttributes( { label } ) }
+					value={ label }
+					onChange={ ( labelValue ) => setAttributes( { label: labelValue } ) }
 					aria-label={ __( 'Navigation Label' ) }
 					maxRows={ 1 }
 				/>
-				<Dropdown
-					contentClassName="wp-block-navigation-menu-item__dropdown-content"
-					position="bottom left"
-					popoverProps={ POPOVER_PROPS }
-					renderToggle={ ( { isOpen, onToggle } ) => (
-						<IconButton
-							icon={ isOpen ? 'arrow-up-alt2' : 'arrow-down-alt2' }
-							label={ __( 'More options' ) }
-							onClick={ onToggle }
-							aria-expanded={ isOpen }
-						/>
-					) }
-					renderContent={ ( { onClose } ) => (
-						<MenuItemActions
-							clientId={ clientId }
-							destination={ attributes.destination }
-							onEditLableClicked={ onEditLableClicked( onClose ) }
-						/>
-					) }
-				/>
 			</div>
 		);
 	} else {
-		content = attributes.label;
+		content = label;
 	}
 	return (
 		<Fragment>
+			<BlockControls>
+				<Toolbar>
+					<ToolbarButton
+						name="link"
+						icon="admin-links"
+						title={ __( 'Link' ) }
+						onClick={ () => setLinkOpen( ! isLinkOpen ) }
+					/>
+					{ isLinkOpen &&
+					<>
+						<URLPopover
+							className="wp-block-navigation-menu-item__inline-link-input"
+						>
+							<URLPopover.LinkEditor
+								value={ url }
+								onChangeInputValue={ ( value ) => setAttributes( { url: value } ) }
+								onKeyPress={ ( event ) => event.stopPropagation() }
+							/>
+						</URLPopover>
+					</>
+					}
+				</Toolbar>
```

Comment:
Does this work? Because the dropdown component has this monitor which closes itself when focus is lost and the URLPopover will steal focus, I have hit the same issue in the MediaFlow component and I had to edit how the dropdown behaves [here](https://github.com/WordPress/gutenberg/pull/16200/files#diff-87d7a98e233541b00690279e85b66a67R148)

I think the media flow is ready and possible that will also solve this problem.

Reply:
It doesn't work as it is, no 😄 
I see that in other instances of LinkEditor a custom function has been used to fix this: [here](https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/image/edit.js#L122) and [here](https://github.com/WordPress/gutenberg/blob/master/packages/format-library/src/link/inline.js#L106). 
I'd say this is another good argument for a toolbar link component that can deal with all these implementation details in a single place.

Reply:
I have now provisionally adopted the same strategy as other LinkEditors I linked to above.